pkg/resource/deploy/step_generator.go

```diff
@@ -1887,6 +1887,34 @@ func (sg *stepGenerator) GenerateRefreshes(
 // registered in the new snapshot. It also generates delete steps for any resources that were marked for deletion
 // because of `destroy` mode.
 func (sg *stepGenerator) GenerateDeletes(targetsOpt UrnTargets, excludesOpt UrnTargets) ([]Step, error) {
+	// If -target was provided to either `pulumi update` or `pulumi destroy` then only delete
+	// resources that were specified.
+	var allowedResourcesToDelete map[resource.URN]bool
+	var forbiddenResourcesToDelete map[resource.URN]bool
+	var err error
+
+	if targetsOpt.IsConstrained() {
+		allowedResourcesToDelete, err = sg.determineAllowedResourcesToDeleteFromTargets(targetsOpt)
+	} else if excludesOpt.IsConstrained() {
+		forbiddenResourcesToDelete, err = sg.determineForbiddenResourcesToDeleteFromExcludes(excludesOpt)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	filter := func(res *resource.State) bool {
```

Comment:
`filter` implies to me that we filter out this resource from the deletes, but the opposite is true here.  Maybe `deleteResource` is a better name, or `allowDeletingResource`?

Reply:
Yeh I was thinking of like `List.filter` where the item is kept if the predicate is true, but maybe `isTargeted`?

Reply:
maybe `isTargetedForDelete` to make it explicit that this is only for deletes?  Though this whole function is about deletes, so `isTarageted` also works for me!